pkg/frontend/nvme_controller.go

```diff
@@ -75,15 +74,22 @@ func NewNvmeTCPTransport(listenAddr string) NvmeTransport {
 	}
 }
 
-func (c *nvmeTCPTransport) Params(_ *pb.NvmeController, nqn string) (spdk.NvmfSubsystemAddListenerParams, error) {
+func (c *nvmeTCPTransport) Params(ctrl *pb.NvmeController, nqn string) (spdk.NvmfSubsystemAddListenerParams, error) {
 	result := spdk.NvmfSubsystemAddListenerParams{}
 	result.Nqn = nqn
-	result.SecureChannel = false
+
+	if *ctrl.Spec.NvmeControllerId < 0 {
+		result.SecureChannel = true
+	} else {
+		result.SecureChannel = false
+	}
+
```

Comment:
ugly hack! should be addressed in https://github.com/opiproject/opi-api/pull/375